libs/application/ui-fields/src/lib/TableRepeaterFormField/TableRepeaterFormField.tsx

```diff
@@ -76,15 +79,29 @@ export const TableRepeaterFormField: FC<Props> = ({
     name: data.id,
   })
 
-  const values = useWatch({ name: data.id, control: methods.control })
   const activeField = activeIndex >= 0 ? fields[activeIndex] : null
   const savedFields = fields.filter((_, index) => index !== activeIndex)
   const tableItems = items.filter((x) => x.displayInTable !== false)
-  const tableHeader = table?.header ?? tableItems.map((item) => item.label)
-  const tableRows = table?.rows ?? tableItems.map((item) => item.id)
   const staticData = getStaticTableData?.(application)
   const canAddItem = maxRows ? savedFields.length < maxRows : true
 
+  let values = useWatch({ name: data.id, control: methods.control })
+  const tableHeader = table?.header ?? tableItems.map((item) => item.label)
+  const tableRows = table?.rows ?? tableItems.map((item) => item.id)
+
+  tableItems.map((item) => {
+    if (item.component === 'nationalIdWithName') {
+      //let table account for name thats being lookup up
+      insert(tableRows, item.id, 'name')
+      insert(tableHeader, item.label, 'Nafn')
+
+      //nationalIdWithName returns an object that we extract entries from and add to values
+      if (values) {
+        values = values.map((value: any) => flattenValueObject(value, item.id))
+      }
+    }
+  })
```

Comment:
Might be a good idea to extract this to a seperate function. We should try to make it clear that this is a solution for components that don't fit nicely into the table repeater. We should also keep an eye on this and possibly update the tableRepeater field to better accommodate components if this gets out of hands.